### ISSUE TITLE
FTL configured with NXDOMAIN still returns NULL result

### ISSUE THREAD

**Issue by bobvandevijver** [pi-hole/FTL#724]
**In raising this issue, I confirm the following (please check boxes, eg [X]) Failure to fill the template will close your issue:**

- [x] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md).
- [x] The issue I am reporting can be *replicated*
- [x] The issue I am reporting isn't a duplicate 

**How familiar are you with the codebase?:**
1
---

**[BUG] Expected Behaviour:**

When I set `BLOCKINGMODE=NXDOMAIN` in `/etc/pihole/pihole-FTL.conf` I expect to receive NXDOMAIN results for blocked hosts, instead of `0.0.0.0`.  

**[BUG] Actual Behaviour:**
With `NXDOMAIN` configured, from the pihole log:
```
Apr  4 12:27:41 dnsmasq[30394]: query[A] e.reddit.com from 192.168.1.36
Apr  4 12:27:41 dnsmasq[30394]: gravity blocked e.reddit.com is NXDOMAIN
Apr  4 12:27:41 dnsmasq[30394]: query[AAAA] e.reddit.com from 192.168.1.36
Apr  4 12:27:41 dnsmasq[30394]: gravity blocked e.reddit.com is NXDOMAIN
```

With `NULL`  configured, from the pihole log:
```
Apr  4 12:28:31 dnsmasq[30806]: query[A] e.reddit.com from 192.168.1.36
Apr  4 12:28:31 dnsmasq[30806]: gravity blocked e.reddit.com is 0.0.0.0
Apr  4 12:28:31 dnsmasq[30806]: query[AAAA] e.reddit.com from 192.168.1.36
Apr  4 12:28:31 dnsmasq[30806]: gravity blocked e.reddit.com is 0.0.0.0
```

However, it always returns `0.0.0.0`: 
Query lookup (results are the same for both modes):
```
bobvandevijver@XPS15-BOB:~/projecten$ nslookup e.reddit.com
Server:         192.168.1.2
Address:        192.168.1.2#53

Name:   e.reddit.com
Address: 0.0.0.0
Name:   e.reddit.com
Address: 0.0.0.0
```

**[BUG | ISSUE] Steps to reproduce:**

- Set blocking mode to `NXDOMAIN`
- See it not being respected

**Device specifics**

Hardware Type: VM
OS: Debian 10

PiHole is running on latest dev branch. This was actually working as expected on the master branch.

_This template was created based on the work of [`udemy-dl`](https://github.com/nishad/udemy-dl/blob/master/LICENSE)._

**Comment by PromoFaux** [comment]
Thanks for the report, unfortunately I am unable to reproduce

![image](https://user-images.githubusercontent.com/1998970/78434230-e3961c80-766d-11ea-9036-a75668f0e271.png)

Are you restarting `FTL` after changing the blocking mode?

**Comment by bobvandevijver** [comment]
Yes, I did after every change. This is almost a clean install, just installed it yesterday but switched to dev for the exclude client possibilities.

![image](https://user-images.githubusercontent.com/1835343/78451024-18609e80-7683-11ea-830f-4d3093079598.png)

**Comment by PromoFaux** [comment]
How strange!

Would you mind providing a debug token please? `pihole -d`

**Comment by bobvandevijver** [comment]
Here is is: https://tricorder.pi-hole.net/zgcyzop3sb

**Comment by PromoFaux** [comment]
Ah, sorry, I misunderstood you're on `dev`, not the beta branch (`release/v5.0`)

I have just switched over to dev and I can reproduce.

![image](https://user-images.githubusercontent.com/1998970/78453264-6e841080-7688-11ea-9792-ffa96e5edac2.png)

I am not sure if this is something that is broken in dev and has since been fixed in the beta branch, or something that has broken since in the dev branch. A lot of the development work for `ftl` has been going into the beta branch for now, with dev getting updated from it from time to time.

For the most stable experience in trying out the features of the next version, it is advised you use the beta branches:

```
echo "release/v5.0" | sudo tee /etc/pihole/ftlbranch
pihole checkout core release/v5.0
pihole checkout web release/v5.0
```

In the mean time, we will look into this on the dev branch (paging @DL6ER )

**Comment by bobvandevijver** [comment]
I just switched to the beta branch (`release/v5.0`), and that indeed resolved the problem. And the group management I switched also works in there, so for me it is fine now :) Although you probably indeed want to check what's happening on dev, I hope it isn't anything that's to much work. 

Anyways, thank you for the swift responses and keep up the good work 👍

**Comment by DL6ER** [comment]
I don't know what is going on on the `development` branches, however, I agree that the issue will be "automatically" resolved when we merged `release/v5.0 -> development`.
